### PR TITLE
Features/landing page cog dropdown

### PR DESCRIPTION
**Pull request by Rossall82**
- Add/edit task notes via the cog on the landing page
- Display hide drop down arrow on task rows depending on existence of task note
- New TaskNote table and related EF Model
- Set up auto mapping for TaskNotes on Index page view model
- Unit tests to cover addition/editing of task notes
- Fix seeded data for 321 process
- Update btn-success instances to btn-primary
- Popper!

### CODE REVIEW

**Comment by UKHOGreg** [comment]
We need to update the Database Design MD.

**Comment by UKHOGreg** [comment]
> 
> 
> We need to update the Database Design MD.

Which I am doing...

File: src/Portal/Portal/Pages/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -25,6 +25,7 @@
     <environment include="Development">
         <script src="~/lib/jquery/jquery.js"></script>
         <script src="~/js/site.js" asp-append-version="true"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
```
Comment (UKHOGreg) [inline src/Portal/Portal/Pages/Shared/_Layout.cshtml]: I think we forgot to LibMan this script.

Comment (UKHOGreg) [reply src/Portal/Portal/Pages/Shared/_Layout.cshtml]: We also need to add it to the 'exclude development' portion as well

**Comment by UKHOGreg** [comment]
Have we updated all TaskSeedData files?

**Comment by Rossall82** [comment]
> Have we updated all TaskSeedData files?

Yes.

**Pull request merged**